source/fe/mapping.cc

```diff
@@ -52,6 +52,33 @@ Mapping<dim, spacedim>::get_vertices(
 
 
 
+template <int dim, int spacedim>
+boost::container::small_vector<Point<spacedim>,
+                               GeometryInfo<dim>::vertices_per_face>
+Mapping<dim, spacedim>::get_vertices(
+  const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+  const typename Triangulation<dim, spacedim>::face_iterator &face) const
+{
+  boost::container::small_vector<Point<spacedim>,
+                                 GeometryInfo<dim>::vertices_per_face>
+    face_vertices;
+
+  const auto  face_nr          = face->index();
+  const auto  face_orientation = cell->combined_face_orientation(face_nr);
```

Comment:
I don't think that is going to work. `face_iterator->index()` (aka `TriaIterator<TriaAccessor<dim - 1, dim, spacedim>>::index()`) is not going to return the local index of a face within a cell, but the index of the face within the collection of all faces. I think you want to hand in the `cell` and the local index of the face, which then allows you to get the orientation. In terms of orientation, I also recommend to check whether this is really the right ordering you are using (cell implying the indices for the face), and not the inverse ordering (face indices relative to the cell). It might be ok, but I do not have the time to look it up precisely and see whether this is correct.

Reply:
Thanks for the fast review. 

I changed the function as suggested and added a test in which I compare the result obtained by the new function and obtained by a detour. Everything seems to work passing the face_no directly :)